src/test/java/edu/cornell/kfs/module/cam/batch/service/impl/CuBatchExtractServiceImplTest.java

```diff
@@ -1,81 +1,191 @@
 package edu.cornell.kfs.module.cam.batch.service.impl;
 
-import edu.cornell.kfs.module.cam.fixture.EntryFixture;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang.StringUtils;
+import org.easymock.Capture;
+import org.easymock.EasyMock;
+import org.junit.Before;
+import org.junit.Test;
 import org.kuali.kfs.gl.businessobject.Entry;
+import org.kuali.kfs.krad.document.Document;
 import org.kuali.kfs.krad.service.BusinessObjectService;
-import org.kuali.kfs.module.cam.batch.service.BatchExtractService;
+import org.kuali.kfs.krad.service.DataDictionaryService;
+import org.kuali.kfs.module.cam.CamsPropertyConstants;
+import org.kuali.kfs.module.purap.PurapConstants;
 import org.kuali.kfs.module.purap.document.PaymentRequestDocument;
 import org.kuali.kfs.module.purap.document.VendorCreditMemoDocument;
-import org.kuali.kfs.sys.ConfigureContext;
-import org.kuali.kfs.sys.context.KualiTestBase;
-import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.rice.krad.bo.BusinessObject;
 
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.List;
+import edu.cornell.kfs.module.cam.CuCamsTestConstants;
+import edu.cornell.kfs.module.cam.fixture.EntryFixture;
+import edu.cornell.kfs.module.purap.document.CuPaymentRequestDocument;
+import edu.cornell.kfs.module.purap.document.CuVendorCreditMemoDocument;
+import edu.cornell.kfs.module.purap.fixture.VendorCreditMemoDocumentFixture;
+
+@SuppressWarnings("deprecation")
+public class CuBatchExtractServiceImplTest {
+
+    private CuBatchExtractServiceImpl cuBatchExtractServiceImpl;
+
+    @Before
+    public void setUp() throws Exception {
+        BusinessObjectService businessObjectService = buildMockBusinessObjectService();
+        DataDictionaryService dataDictionaryService = buildMockDataDictionaryService();
+        cuBatchExtractServiceImpl = new CuBatchExtractServiceImpl();
+        cuBatchExtractServiceImpl.setBusinessObjectService(businessObjectService);
+        cuBatchExtractServiceImpl.setDataDictionaryService(dataDictionaryService);
+    }
+
+    @Test
+    public void testFindCreditMemoDocument() {
+        VendorCreditMemoDocument vendorCreditMemoDocument = cuBatchExtractServiceImpl.findCreditMemoDocument(EntryFixture.VCM_TWO.createEntry());
+        assertNotNull("vendor credit memo should have been non-null", vendorCreditMemoDocument);
+        assertEquals("Wrong credit memo document was retrieved", CuCamsTestConstants.DOC_5686500, vendorCreditMemoDocument.getDocumentNumber());
+        
+        vendorCreditMemoDocument = cuBatchExtractServiceImpl.findCreditMemoDocument(EntryFixture.VCM_ONE.createEntry());
```

Comment:
I think it's just retrieving a different CM document than the one above it. Given that this doesn't add much value, though, I'll refactor or remove this second retrieval accordingly.